ledger/acctupdates.go

```diff
@@ -1487,6 +1493,44 @@ func (au *accountUpdates) upgradeDatabaseSchema3(ctx context.Context, tx *sql.Tx
 	return 4, nil
 }
 
+// upgradeDatabaseSchema4 does not change the schema but migrates data:
+// remove empty AccountData entries from accountbase table
+func (au *accountUpdates) upgradeDatabaseSchema4(ctx context.Context, tx *sql.Tx) (updatedDBVersion int32, err error) {
+
+	queryAddresses := au.balancesTrie != nil
```

Comment:
check `au.catchpointInterval == 0` instead, since you can't use the `au.balancesTrie` at this point of initialization.

Reply:
done